numba/core/decorators.py

```diff
@@ -24,7 +24,7 @@
                                  "positional argument.")
 
 def jit(signature_or_function=None, locals={}, cache=False,
-        pipeline_class=None, boundscheck=False, **options):
+        pipeline_class=None, boundscheck=None, **options):
```

Comment:
This has ended up being not particularly neat. The value for "boundscheck" is transmitted via `targetoptions` and actually originates here, and is actually captured in this function:
https://github.com/numba/numba/blob/1041fa6ee8430471da99b54b3428a673033e7e44/numba/core/decorators.py#L154
it then propagates.

To permit changing behaviour when `debug` is set a 3-state is needed, `True/False` as potentially set by a user and `None` for "not set, let it be influenced by `debug`".